model/project_ref_test.go

```diff
@@ -1641,6 +1641,80 @@ func TestCreateNewRepoRef(t *testing.T) {
 	assert.NotContains(t, scope.Resources, doc1.Id)
 }
 
+func TestGitHubDynamicTokenPermissions(t *testing.T) {
+	perms := GitHubDynamicTokenPermissions{
+		GitHubDynamicTokenPermission{
+			Requesters: []string{"requester1", "requester2"},
+			Permissions: map[string]string{
+				"checks": "write",
+				"issues": "read",
+			},
+		},
+		GitHubDynamicTokenPermission{
+			Requesters: []string{"requester3"},
+			Permissions: map[string]string{
+				"content": "read",
+				"fake":    "write",
+			},
+		},
+	}
+
+	t.Run("Invalid requester specified", func(t *testing.T) {
+		assert.Nil(t, perms.Get("requester0"))
+	})
+
+	t.Run("Valid requester specified", func(t *testing.T) {
+		perm := perms.Get("requester1")
+		require.NotNil(t, perm)
+		assert.Equal(t, "write", perm.Permissions["checks"])
+		assert.Equal(t, "read", perm.Permissions["issues"])
+
+		perm = perms.Get("requester2")
+		require.NotNil(t, perm)
+		assert.Equal(t, "write", perm.Permissions["checks"])
+		assert.Equal(t, "read", perm.Permissions["issues"])
+
+		perm = perms.Get("requester3")
+		require.NotNil(t, perm)
+		assert.Equal(t, "read", perm.Permissions["content"])
+		assert.Equal(t, "write", perm.Permissions["fake"])
+	})
+}
+
+func TestGitHubDynamicTokenPermission(t *testing.T) {
+	t.Run("Valid permissions", func(t *testing.T) {
+		perm := GitHubDynamicTokenPermission{
+			Requesters: []string{"requester1", "requester2"},
+			Permissions: map[string]string{
+				"checks": "write",
+				"issues": "read",
+			},
+		}
+
+		githubPerms, err := perm.AsGitHubInstallationPermissions()
+		require.NoError(t, err)
+		require.NotNil(t, githubPerms.Checks)
+		assert.Equal(t, "write", *githubPerms.Checks)
+		require.NotNil(t, githubPerms.Issues)
+		assert.Equal(t, "read", *githubPerms.Issues)
+	})
+
+	t.Run("With some invalid permissions", func(t *testing.T) {
+		perm := GitHubDynamicTokenPermission{
+			Requesters: []string{"requester3"},
+			Permissions: map[string]string{
+				"contents": "read",
+				"fake":     "write",
```

Comment:
Does github not error in this case? I wonder if maybe we should error if the resulting struct has less than what the user requested. They may have misspelled "contents" to "content" or something like that and then silently fail.

Reply:
I'll switch the decoder over to error if it finds anything invalid and update this test!